jerry-core/jmem/jmem-heap.c

```diff
@@ -411,11 +406,60 @@ jmem_heap_alloc_block (const size_t size)
   }
 
   JERRY_ASSERT (data_space_p == NULL);
+  return data_space_p;
+} /* jmem_heap_gc_and_alloc_block */
+
+
+/**
+ * Allocation of memory block, running 'try to give memory back' callbacks, if there is not enough memory.
+ *
+ * Note:
+ *      if after running the callbacks, there is still not enough memory, engine is terminated with ERR_OUT_OF_MEMORY.
```

Comment:
IMHO this sounds better:
"if there is still not enough memory after running the callbacks, then the engine will be terminated with ERR_OUT_OF_MEMORY."